File: frontend_agricare/lib/models/community_chat/message.dart
class MessageModel {
  final String sender;
  final String? message;
  final String? fileUrl;

  MessageModel({
    required this.sender,
    this.message,
    this.fileUrl,
  });

  factory MessageModel.fromJson(Map<String, dynamic> json) {
    return MessageModel(
      sender: json["sender"],
      message: json["message"],
      fileUrl: json["fileUrl"],
    );
  }
}
  

File: Backend/src/models/community_chat/Message.js
import mongoose from "mongoose";

const messageSchema = new mongoose.Schema({
  roomId: { 
    type: mongoose.Schema.Types.ObjectId, 
    ref: "ChatRoom", 
    required: true 
  },
  sender: { 
    type: String, 
    required: true 
  },
  message: { 
    type: String 
  },
  senderName: { 
    type: String 
  },
  fileUrl: { 
    type: String, 
    default: null 
  },
  readBy: [{ 
    type: String 
  }],
  createdAt: { 
    type: Date, 
    default: Date.now 
  }
});

export default mongoose.model("Message", messageSchema);